sql/src/main/java/io/crate/operation/projectors/FileWriterCountCollector.java

```diff
@@ -156,38 +158,49 @@ private RowWriter initWriter() {
         }
     }
 
+    private void closeWriterAndOutput() {
+        try {
+            if (rowWriter != null) {
+                rowWriter.close();
+            }
+        } catch (IOException e) {
+
+        }
+    }
+
     @Override
-    public Result setNextRow(Row row) {
-        rowWriter.write(row);
-        counter.incrementAndGet();
-        return Result.CONTINUE;
+    public Supplier<AtomicLong> supplier() {
+        return () -> new AtomicLong(0);
     }
 
     @Override
-    public void finish(RepeatHandle repeatHandle) {
-        if (closeWriterAndOutput()) return;
+    public BiConsumer<AtomicLong, Row> accumulator() {
+        return this::onNextRow;
+    }
 
-        downstream.setNextRow(new Row1(counter.get()));
-        downstream.finish(RepeatHandle.UNSUPPORTED);
+    private void onNextRow(AtomicLong container, Row row) {
+        rowWriter.write(row);
+        container.incrementAndGet();
```

Comment:
`Collectors.summingLong` internally uses a 1 element long array as state - we could use that pattern as well to prevent the synchronization overhead of the AtomicInteger which we don't need here.